engine/namegen.ml

```diff
@@ -404,14 +404,13 @@ let rename_bound_vars_as_displayed avoid env c =
 
 let h_based_elimination_names = ref false
 
-let use_h_based_elimination_names () =
-  !h_based_elimination_names && Flags.version_strictly_greater Flags.V8_4
```

Comment:
Actually, this option was never activated by default because I thought it would have led to too much incompatibilities. Moreover, we can do even better in naming hypotheses produced by destruct/induction. So, not only I see this option as only a step towards a big picture, but I don't think it is reasonable to change the naming model until we have a way to globally change the tactic language using commands to invent such as "Require Tactics (version 8.5)" or similar.

(Note: the policy that some people advocate of naming everything by hand is also fine to me!)

In particular, I would not deprecate it.

There is another option in this situation, it is the "Structural Injection" which is off by default.